src/main/kotlin/no/nav/integrasjon/ldap/LDAPGroup.kt

```diff
@@ -159,100 +170,208 @@ class LDAPGroup(private val config: FasitProperties) :
-                                    add(Attribute("cn", groupName))
-                                    add(Attribute("sAMAccountName", groupName))
-
-                                    resolveUserDN(creator).let { userDN ->
-                                        if (groupName.startsWith(KafkaGroupType.MANAGER.prefix) &&
-                                                userDN.isNotEmpty())
-                                            add(Attribute(config.ldapGrpMemberAttrName, userDN))
-                                    }
-                                })
-                                .also { req -> log.info { "Create group request: $req" } }
-                ).simplify()
+        if (exists)
+            LDAPResult(ResultCode.ENTRY_ALREADY_EXISTS_INT_VALUE, ResultCode.ENTRY_ALREADY_EXISTS).simplify()
+        else
+            ldapConnection.add(
+                AddRequest(
+                    DN(config.groupDN(groupName)),
+                    newGroupAttr.toMutableList().apply {
+                        add(Attribute("cn", groupName))
+                        add(Attribute("sAMAccountName", groupName))
+
+                        resolveUserDN(creator).let { userDN ->
+                            if (groupName.startsWith(KafkaGroupType.MANAGER.prefix) &&
+                                userDN.isNotEmpty()
+                            )
+                                add(Attribute(config.ldapGrpMemberAttrName, userDN))
+                        }
+                    })
+                    .also { req -> log.info { "Create group request: $req" } }
+            ).simplify()
 
     private fun deleteKafkaGroup(exists: Boolean, groupName: String): SLDAPResult =
-            if (!exists)
-                LDAPResult(ResultCode.NO_SUCH_OBJECT_INT_VALUE, ResultCode.NO_SUCH_OBJECT).simplify()
-            else
-                ldapConnection.delete(
-                    DeleteRequest(
-                            DN(config.groupDN(groupName))
-                    ).also { req -> log.info { "Delete group request: $req" } }
+        if (!exists)
+            LDAPResult(ResultCode.NO_SUCH_OBJECT_INT_VALUE, ResultCode.NO_SUCH_OBJECT).simplify()
+        else
+            ldapConnection.delete(
+                DeleteRequest(
+                    DN(config.groupDN(groupName))
+                ).also { req -> log.info { "Delete group request: $req" } }
             ).simplify()
 
     fun getKafkaGroupMembers(groupName: String) = getMembersInKafkaGroup(groupName)
 
+    fun getGroupInGroupMembers(groupName: String) = getMembersInGroupInGroup(getCNFromDN(groupName))
+
+    // REGEX approved Trong?
+    private fun getCNFromDN(dNGroupName: String) = """([^,]*)""".toRegex().find(dNGroupName)!!.value.replace("cn=", "")
+
+    data class GroupInGroupException(val msg: String) : Exception(msg)
+
+    data class UserNotAllowedException(val msg: String) : Exception(msg)
+
+    data class UserNotFound(val msg: String) : Exception(msg)
```

Comment:
Burde vi gjøre noe annet enn å kaste exceptions her?

Reply:
hva er forslaget? 👍

Reply:
Forslaget mitt er å flytte logikken for tilgangstyring ut, har ikke sett nok i koden til å ha en god løsning på det

Reply:
Skal jeg gjøre dette i denne issue:en? @TheUnnamedDude

Reply:
Lager et forslag.. 🗡

Reply:
Lagd et forslag nå 💯